prboom2/src/r_draw.c

```diff
@@ -587,3 +591,14 @@ int R_GetFuzzPos()
 {
   return fuzzpos;
 }
+
+void R_ResetFuzzCol(int rows)
+{
+  R_ResetColumnBuffer(); // Flush current columns before changing fuzzpos
```

Comment:
Try to avoid redundant comments like this:
```c
A(); // call A before B
B();
```
The code is pretty clear here :^)

Reply:
Eh, it could look more like
`SeeminglyUnrelatedRoutine();
ChangingFuzzpos();`
to someone not as familiar with doom's software renderer. Maybe the comment could be better, such as "Make sure columns are drawn before changing fuzzpos"

Reply:
That comment wouldn't have any meaning either. The purpose of a comment would be to explain _why_ something is happening. It's very clear _what_ is happening. When the _what_ is unclear, it means there's a problem in the names of things or in how the code is organized generally (i.e., a comment is the wrong solution). I see a function purporting to reset the fuzz column, and it calls a function that resets the column buffer. That seems reasonable to me - of course that doesn't explain everything that happens, but the reader can choose to follow the function if they want more details.

Reply:
The proposed comment does explain the why. "Make sure columns are drawn before changing fuzzpos"

Reply:
That is merely stating what the code does - as a reader I can usually assume that if A happens before B it is because A should happen before B. A comment comes in when it's unclear why that thing needs to happen at all or why it needs to happen at that moment, neither of which are covered by the comment here. Another case would be if there is something shockingly unexpected that you need to communicate to the reader to ensure they don't break it in the future (e.g., that there is a memory hack used on a struct and the position of some element must not change). Regardless of the semantics of "why" I don't think a comment is needed in this case.

Reply:
It's stating it resets the column buffer to make sure the columns are drawn before changing fuzzpos, which explains why it's happening.
It can be unclear what "ResetColumnBuffer" means exactly, so that's why I added the comment.

Reply:
This is what I meant above when I said if the core problem is actually an unclear name, the solution is to change the name or how the code is structured. In other words, to instead call a function like `R_DrawColumns`. But in this case I'm not suggesting to do that change, because we are in the context of the software renderer and should maintain the nomenclature, at least for this PR. Looking through that lens, the code makes sense as it is. Basically this isn't the place to solve the knowledge gap, because then we would need a thousand comments 😄

Reply:
Ah, you wanna remove it now so that in the future, when the function's called something more descriptive, the comment isn't mostly redundant. I can remove it in anticipation for that change.